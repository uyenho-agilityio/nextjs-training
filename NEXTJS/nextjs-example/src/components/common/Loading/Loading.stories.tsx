// Lib
import Storybook from '@storybook/react';

// Component
import { Loading } from './Loading';

export default {
  title: 'Components/Loading',
  component: Loading,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as Storybook.ComponentMeta<typeof Loading>;

const Template: Storybook.ComponentStory<typeof Loading> = args => <Loading {...args} />;

export const Default = Template.bind({});
export const WithColor = Template.bind({});
WithColor.args = {
  color: 'secondary-cl',
  emptyColor: 'gray.200',
};
