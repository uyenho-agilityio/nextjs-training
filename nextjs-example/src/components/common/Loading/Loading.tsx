// Libs
import { Container, Spinner, SpinnerProps as SpinnerPropsChakra } from '@chakra-ui/react';
import React from 'react';

type SpinnerProps = {
  variant?: 'default' | 'center';
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  color?: string;
  thickness?: string;
} & SpinnerPropsChakra;

export const Loading: React.FC<SpinnerProps> = ({
  variant = 'default',
  size = 'lg',
  color = 'primary-cl',
  thickness = '3px',
  ...props
}): JSX.Element => {
  return (
    <Container variant={variant}>
      <Spinner size={size} color={color} thickness={thickness} {...props} />
    </Container>
  );
};
