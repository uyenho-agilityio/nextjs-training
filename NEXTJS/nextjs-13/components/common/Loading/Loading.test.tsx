// Lib
import { render } from '@testing-library/react';

// Component
import { Loading } from './Loading';

describe('Loading render', () => {
  test('should render loading component', () => {
    const { container } = render(<Loading />);
    expect(container).toMatchSnapshot();
  });
});
