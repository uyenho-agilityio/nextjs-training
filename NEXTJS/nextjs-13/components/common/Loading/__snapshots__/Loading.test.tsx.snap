// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading render should render loading component 1`] = `
<div>
  <div
    class="chakra-container css-0"
  >
    <div
      class="chakra-spinner css-19krcf2"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
